File: 18_정지윤/2108.java
import javax.swing.*;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    private void solve() {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))){
            int N = Integer.parseInt(br.readLine());            //수의 개수 N은 홀수
            int[] number = new int[N];
            int[] turn = new int[8001];
            List<Integer> list = new ArrayList<>();

            int first, second, third = 0, forth;
            double sum = 0;

            for (int i = 0; i < N; ++i) {
                int num = Integer.parseInt(br.readLine());
                number[i] = num;
                sum += num;

                turn[4000+num]++;
            }

            first = (int)(Math.round(sum/N));//첫번째 출력

            Arrays.sort(number);
            second = number[N/2];//두번째 출력

            int max = 0;
            boolean flag = false;

            for(int i = 0; i < turn.length; i++) {
                if(turn[i] > max) {
                    max = turn[i];
                    third = i - 4000;
                    flag = true;
                } else if(turn[i] == max && flag == true) {
                    third = i - 4000;
                    flag = false;
                }
            }

            forth = number[N-1] - number[0];

            System.out.println(first);
            System.out.println(second);
            System.out.println(third);
            System.out.println(forth);

        }catch (IOException e) {
            e.printStackTrace();
        }
    }
    public static void main(String[] args){
        new Main().solve();
    }
}

File: 03_이하정/2108.py
from collections import Counter

n=int(input())
k=[]
for i in range(n):
    k+=[int(input())]

k.sort()
# 산술평균
print(round(sum(k)/len(k)))
# 중앙값 (n은 항상 홀수)
if len(k)==1:
    print(k[0])
else:
    print(k[len(k)//2])
# 최빈값
d=Counter(k) # 리스트의 요소들의 개수를 세어, 딕셔너리 형태로 반환 {문자:개수}
m=[n for n in d.keys() if d[n] == max(d.values())] # 해당 요소가 최빈값인 경우 그 요소를 리스트로 저장
if len(m)==1:
    print(m[0])
else: # 최빈값이 2개 이상인 경우, 2번째로 작은 값 출력
    m.remove(min(m))
    print(min(m))
# 범위 
print(k[-1]-k[0])